products/BellHybrid/apps/common/src/widgets/TimeSetSpinnerVertical.cpp

```diff
@@ -61,22 +80,30 @@ namespace gui
 
         switch (newFormat) {
         case utils::time::Locale::TimeFormat::FormatTime12H: {
-            auto hours = std::chrono::hours(hour->value());
-            hour->set_range(UIntegerSpinnerFixed::range{
-                time::Locale::min_hour_12H_mode, time::Locale::max_hour_12H_mode, hourStep});
+            auto hourFirst = hour_first->getText();
+            auto hourSecond = hour_second->getText();
+            auto hoursStr = hourFirst + hourSecond;
+            auto hours = std::chrono::hours(std::stoi(hoursStr.c_str()));
```

Comment:
all those variables are used in both cases in this switch. Maybe we can set them before switch?

Reply:
Updated